bootcfg/api/handlers.go

```diff
@@ -25,3 +30,37 @@ func logRequests(next http.Handler) http.Handler {
 	}
 	return http.HandlerFunc(fn)
 }
+
+// selectGroup selects the Group whose selectors match the query parameters,
+// adds the Group to the ctx, and calls the next handler. The next handler
+// should handle a missing Group.
```

Comment:
nit: having comments here is great for GoDoc. It's also nice to have them in the proto file so the entire API is human parsable with 1 file. Maintaining 2 sets of identical comments is not good, so I propose putting them in just the proto file b/c the actual server implementation methods are unlikely to be dependencies of anything.